### PR TITLE
Employee loan fixes

### PR DESCRIPTION
**Pull request by rohitwaghchaure**
- Added child table in salary slip to link salary slip and employee loan

- Fixed issue of interest amount, employee loan amount  not booked in respective interest account and employee loan account

- Added patch to update employee loan data in child table

Employee loan details in salary slip
screen shot 2017-11-09 at 12 47 09 pm

Fixed https://github.com/frappe/erpnext/issues/11366

### CODE REVIEW

File: erpnext/hr/doctype/employee_loan_detail/employee_loan_detail.json [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+ "image_view": 0, 
+ "in_create": 0, 
+ "is_submittable": 0, 
+ "issingle": 0, 
+ "istable": 1, 
+ "max_attachments": 0, 
+ "modified": "2017-11-08 16:33:34.299724", 
+ "modified_by": "Administrator", 
+ "module": "HR", 
+ "name": "Employee Loan Detail",
```
Comment (nabinhait) [inline erpnext/hr/doctype/employee_loan_detail/employee_loan_detail.json]: Why not "Salary Slip Loan"?

File: erpnext/hr/doctype/salary_slip/salary_slip.json [hunk 2]
```diff
@@ -1275,11 +1288,73 @@
    "unique": 0
   }, 
   {
+   "allow_bulk_edit": 0, 
+   "allow_on_submit": 0, 
+   "bold": 0, 
+   "collapsible": 0, 
+   "columns": 0, 
+   "fieldname": "loan_details",
```
Comment (nabinhait) [inline erpnext/hr/doctype/salary_slip/salary_slip.json]: why not "loans"?

File: erpnext/hr/doctype/employee_loan_detail/employee_loan_detail.json [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+   "no_copy": 0, 
+   "options": "", 
+   "permlevel": 0, 
+   "precision": "", 
+   "print_hide": 0, 
+   "print_hide_if_no_value": 0, 
+   "read_only": 1, 
+   "remember_last_selected_value": 0, 
+   "report_hide": 0, 
+   "reqd": 0,
```
Comment (nabinhait) [inline erpnext/hr/doctype/employee_loan_detail/employee_loan_detail.json]: set required columns in this table

**Review by nabinhait — CHANGES_REQUESTED** [review]
Add a gif file to show it is working

**Comment by nabinhait** [comment]
closing it for now

**Comment by rohitwaghchaure** [comment]
@nabinhait Fixed!

**Review by nabinhait — APPROVED** [review]

**Pull request merged**